internal/diag.go

```diff
@@ -106,49 +112,56 @@ func Run(params Params) error {
 		},
 	})
 
+	var operatorLabels []labels.Set
 	operatorVersions := make([]*version.Version, 0, len(params.OperatorNamespaces))
-
 	for _, ns := range params.OperatorNamespaces {
-		logger.Printf("Extracting Kubernetes diagnostics from %s\n", ns)
+		// Find the label in use by operator in this namespace and add this to
+		// the set of filters to ensure we always retrieve the objects
+		// associated with the ECK operator.
+		operatorLabel, err := getOperatorLabel(kubectl, ns)
+		if err != nil || operatorLabel == nil {
+			logger.Printf("Could not find label corresponding to ECK Operator in namespace %s: potentially not including operator data in diagnostics", ns)
+		} else {
+			operatorLabels = append(operatorLabels, operatorLabel)
+		}
 
 		operatorVersions = append(operatorVersions, detectECKVersion(clientSet, ns, params.ECKVersion))
-
-		zipFile.Add(getResources(kubectl.GetByLabel, ns, filters.Filters{}, []string{
-			"statefulsets",
-			"pods",
-			"services",
-			"configmaps",
-			"events",
-			"networkpolicies",
-			"controllerrevisions",
-			"serviceaccount",
-		}))
-
-		zipFile.Add(map[string]func(io.Writer) error{
-			archive.Path(ns, "secrets.json"): func(writer io.Writer) error {
-				return kubectl.GetMeta("secrets", ns, writer)
-			},
-		})
-
-		// Filters is intentionally empty here, as label filtering doesn't apply to
-		// pods in the operator namespace.
-		if err := kubectl.Logs(ns, "", filters.Filters{}, zipFile.Create); err != nil {
-			zipFile.AddError(err)
-		}
 	}
 
 	maxOperatorVersion := max(operatorVersions)
 	logVersion(maxOperatorVersion)
 
+	allNamespaces := sets.New(params.ResourcesNamespaces...)
+	allNamespaces.Insert(params.OperatorNamespaces...)
+
 LOOP:
-	for _, ns := range params.ResourcesNamespaces {
+	for ns := range allNamespaces {
 		select {
 		case <-stopCh:
 			break LOOP
 		default:
 		}
+
+		logsLabels := []string{
+			"common.k8s.elastic.co/type=elasticsearch",
+			"common.k8s.elastic.co/type=kibana",
+			"common.k8s.elastic.co/type=apm-server",
+			// the below were introduced in later version but label selector will just return no result:
+			"common.k8s.elastic.co/type=enterprise-search", // 1.2.0
+			"common.k8s.elastic.co/type=beat",              // 1.2.0
+			"common.k8s.elastic.co/type=agent",             // 1.4.0
+			"common.k8s.elastic.co/type=maps",              // 1.6.0
+		}
+
+		selectors := make([]labels.Selector, len(operatorLabels))
+		for i, label := range operatorLabels {
+			selectors[i] = label.AsSelector()
+			logsLabels = append(logsLabels, label.AsSelector().String())
+		}
+		namespaceFilters := params.Filters.WithSelectors(selectors)
```

Comment:
I think we can now simplify a bit more by moving this block outside of the loop and add the operator label selector to `params.Filters` and `logsLabel` on the fly when iterating through operator namespaces.

Reply:
We could even avoid constructing and iterating over `operatorLabels` but I think it's a little clearer that way.